src/bin/pgcopydb/ld_stream.c

```diff
@@ -767,8 +772,54 @@ streamRotateFile(LogicalStreamContext *context)
 		return true;
 	}
 
-	/* compute the WAL filename that would host the current LSN */
-	XLByteToSeg(context->cur_record_lsn, segno, context->WalSegSz);
+	/*
+	 * calculate the lsn based upon which we determine the walFileName
+	 */
+	uint64_t lsn;
```

Comment:
After countless minutes of thinking I couldn't think of a better name :)

Reply:
At least expand the comment to explain the logic behind not always using the cur_record_lsn for the computation of the WAL filename.

What I had in mind personally was to compute the WAL filename anyway and then ignore it when it's in the past. That can be determined by a `strcmp` or the filename + 8 (skip the timeline). That said working with LSN is probably better here (we know the timeline can't change without a full reconnect anyway).

Reply:
Variable name candidate: `jsonFileLSN` ; allows the comment to explain that the WAL LSN is monotonous but the logical decoding protocol only sends transactions at commit time, and then concurrent transaction interleaving means that we don't always receive the LSNs in order.